File: algo/university/sorting algorithms/python/bubbleSort.py
from helpers.swap import swap

def bubbleSort(arr:list[float])->None:
    length = len(arr)
    for i in range(length - 1):
        isSorted = True
        for j in range(0, length - i - 1):
            if arr[j] > arr[j + 1]:
                swap(arr, j + 1 ,j)
                isSorted = False
        if isSorted:
            break

File: ads/sorting algorithms/c/bubbleSort.c
void bubbleSort(float arr[], int length)
{
    for (int i = 0; i < length - 1; i++)
    {
        bool isSorted = true;
        for (int j = 0; j < length - i - 1; j++)
            if (arr[j] > arr[j + 1])
            {
                swap(&arr[j], &arr[j + 1]);
                isSorted = false;
            }
        if (isSorted)
            break;
    }
}
